setuptools/__init__.py

```diff
@@ -224,5 +224,25 @@ def findall(dir=os.curdir):
     return list(files)
 
 
+def get_version(path, field='__version__'):
+    """
+    Get version information from a text file. Version is extracted
+    from "key = value" format with key matched at the beginning of
+    a line and specified by `field` parameter. Returns string.
+    """
+    version_file = os.path.abspath(path)
+    # Using binary matching to avoid possible encoding problems
+    # when reading arbitrary text files
+    if type(field) is not bytes:
+        field = field.encode('utf-8')
```

Comment:
This encode should happen unconditionally.

Reply:
What if it is already `bytes`?